game/game/playercontrol.cpp

```diff
@@ -386,11 +386,21 @@ void PlayerControl::marvinF8(uint64_t dt) {
   float s   = std::sin(rot), c = std::cos(rot);
 
   Tempest::Vec3 dp(s,0.8f,-c);
-  pos += dp*6000*float(dt)/1000.f;
+  dp *= 6000*float(dt)/1000.f;
 
   pl.changeAttribute(ATR_HITPOINTS,pl.attribute(ATR_HITPOINTSMAX),false);
   pl.changeAttribute(ATR_MANA,     pl.attribute(ATR_MANAMAX),     false);
   pl.clearState(false);
+
+  if (pl.isSwim()) {
+    auto waterRay = pl.world().physic()->waterRay(pos);
+    auto res      = pl.world().physic()->ray(pos,{pos.x,20000,pos.z});
+    auto gl       = pl.guild();
+    float chest   = float(pl.world().script().guildVal().water_depth_chest[gl]);
+    if ((waterRay.hasCol && pos.y+chest>waterRay.wdepth) || (!res.hasCol && !waterRay.hasCol))
```

Comment:
What this checks for? Why straight overriding swim/dive flag is not enough? Why 20000?

Reply:
Overriding ports you straight to the surface, while vanilla allows underwater use. 20000 is `DynamicWorld::worldHeight`. I made it a constant now as well.